components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/TrackingProtectionExceptionFileStorage.kt

```diff
@@ -158,6 +158,8 @@ internal class TrackingProtectionExceptionFileStorage(
             trackingExceptions.forEach {
                 storage.setPermission(it, VALUE_DENY)
             }
+        }.finally_ {
```

Comment:
I'm not sure what is `finally_`, but you could execute the callback after all exceptions are denied on line 159.

```kotlin
        storage.allPermissions.accept { permissions ->
            val trackingExceptions = permissions.filterTrackingProtectionExceptions()
            trackingExceptions.forEach {
                storage.setPermission(it, VALUE_DENY)
            }
            onRemoveEnd.invoke()
        }
```